### DISCUSSION TITLE
How to update citation key for existing item in library?

### DISCUSSION THREAD

**Discussion by chaicurioquest** [retorquere/zotero-better-bibtex#2974] [Q&A] [answered]
Hi,
I am quite lost in BBT while updating citation key. I tried to get help in earlier discussion forum and https://retorque.re/zotero-better-bibtex/citing/. I have library collection with existing keys before installation BBT for e.g. "comberAmorphoussiliconFieldeffect1979".  I have given default citation formula "auth.lower + shorttitle(3,3) + year". Furthermore, I have tried with refresh the right-click on item and select Better BibTex -->  Refresh BibTex key with and without pinning. The key is not getting updated. Any help?

And, is there any way to update the all existing library items citation key into new format? Thank you.

**Comment by retorquere** [comment]
I can't say what's going on without a debug log. Reproduce and send a log please.

> And, is there any way to update the all existing library items citation key into new format? Thank you.

Select all, right-click, BBT -> Refresh.

**Comment by chaicurioquest** [comment]
Thanks for the reply.  
I have got following after the refresh, I think it worked. 
![image](https://github.com/user-attachments/assets/b125478d-ee40-4bf6-8a12-cd8c9d0d194e)

My citation formula is `auth.lower + shorttitle(1,1).lower + year. ` Some of the generated keys are lengthy (ref image above) to use in latex with \cite{}. Can you suggest any optimized citation formula? I have few queries (Excuse me for asking a basic questions.  I have got confused with steps) 1. What is the purpose of Pin and Unpin BibTex and Automatically pin the citation key after under Edit→ Settings→ Better BibTex ? 2. I am trying to get the citation details including notes and annotation through zotxt plugin to use in Emacs Org. Is the default BBT setting sufficient for this purpose? . Thx

**Reply by retorquere** [reply, marked as answer]
> Can you suggest any optimized citation formula?

"optimized" is very personal. If you just want it to be shorter, you can do something like

```
auth.lower + shorttitle(1,1).lower.substring(n=5) + year
```

> What is the purpose of Pin and Unpin BibTex and Automatically pin the citation key after 

Pinning the citation key does two things:

1. the key will stay at its current value regardless of further changes to the item, and
2. the citation key becomes shared with other places your library is synced to (people, but also eg the sync with Overleaf)

auto-pin-after just means that after a few seconds after the last change/creation, BBT will pin the key for you so you don't have to do this manually.

> I am trying to get the citation details including notes and annotation through zotxt plugin. Is the default BBT setting sufficient for this purpose? 

I haven't used zotxt, so I cannot answer this.

**Reply by chaicurioquest** [reply]
Thank you
